### DISCUSSION TITLE
Mobile manipulator motion planning and IK using CuRobo

### DISCUSSION THREAD

**Discussion by im-renpei** [NVlabs/curobo#88] [Software Q&A] [answered]
Hi everyone,

I would like to seek some advice on implementing a mobile robot using CuRobo for motion planning and collision-free IK computing. In detail, the robot is equipped with a chassis that can translate (x-axis and y-axis) and rotate around the z-axis; a lift that can control the translation and rotation (in z-axis) of the robot's body (e.g., rising and falling); and two 7-DoF robotic arms on the left and right sides.

In this case, I have several questions:

1. Can CuRobo realize cooperative motion planning for the two robotic arms?
2. For a single robot arm, when performing collision-free IK, can I specify the target to be set in any coordinate system? For example, when controlling a robotic arm to reach a target pose in the chassis coordinate system, will the IK solution include the values (translate and rotate) of the lift? Or can I only calculate the IK of the robotic arm?

Thank you in advance for any insights you can provide.

**Comment by im-renpei** [comment]
Hi @balakumar-s , do you have any suggestions?

**Comment by balakumar-s** [comment, marked as answer]
You can definitely generate motions for your whole robot with cuRobo. 

Answers to your questions:
1. Yes, check this isaac sim demo for an example: https://curobo.org/source/getting_started/2b_isaacsim_examples.html#multi-arm-reacher
You can set targets for each end-effector of your robot and cuRobo will generate motions for the whole robot to make the two end-effectors reach their respective targets.
2. I am a little unclear on the question. cuRobo will output the full joint configuration solution for your robot which will include the values for the lift system. You cannot dynamically change the coordinate system, you can either transform from any coordinate system to a predefined frame or have two instances of cuRobo (one in world frame, one in chassis frame). If you set the coordinate system to be the base of the arm, then you will get IK only for the arm joints.

**Reply by balakumar-s** [reply]
@im-renpei is this a publicly available robot? 

If yes, we can look into creating a tutorial on dual-arm mobile robot motion generation with cuRobo.
